src/test/unit/algorithms/SDRClassifierTest.cpp

```diff
@@ -0,0 +1,295 @@
+/* ---------------------------------------------------------------------
+ * Numenta Platform for Intelligent Computing (NuPIC)
+ * Copyright (C) 2016, Numenta, Inc.  Unless you have an agreement
+ * with Numenta, Inc., for a separate license for this software code, the
+ * following terms and conditions apply:
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ * ---------------------------------------------------------------------
+ */
+
+/** @file
+ * Implementation of unit tests for SDRClassifier
+ */
+
+#include <iostream>
+#include <sstream>
+
+#include <gtest/gtest.h>
+
+#include <nupic/algorithms/ClassifierResult.hpp>
+#include <nupic/algorithms/SDRClassifier.hpp>
+#include <nupic/math/StlIo.hpp>
+#include <nupic/types/Types.hpp>
+#include <nupic/utils/Log.hpp>
+
+using namespace std;
+using namespace nupic;
+using namespace nupic::algorithms::cla_classifier;
+using namespace nupic::algorithms::sdr_classifier;
+
+namespace
```

Comment:
Good question. Here's an example...

Without an anonymous namespace,

~~~
TEST(ExtendedTemporalMemoryTest, ActivateCorrectlyPredictiveCells)`
~~~

is compiled to
~~~
unit_tests`ExtendedTemporalMemoryTest_ActivateCorrectlyPredictiveCells_Test::TestBody
~~~

So the macro doesn't do much magic. Here it's gonna be

~~~
unit_tests`SDRClassifierTest_Basic_Test::ClassBody
~~~